messaging/src/main/java/org/axonframework/commandhandling/gateway/AbstractRetryScheduler.java

```diff
@@ -0,0 +1,175 @@
+
+    /**
+     * This is the entrypoint of the {@link RetryScheduler}. This default implementation checks if the last failure was
+     * transient, and if so reschedules a command dispatch.
+     *
+     * @param commandMessage  The Command Message being dispatched
+     * @param lastFailure     The last failure recorded for this command
+     * @param failures        A condensed view of all known failures of this command. Each element in the array
+     *                        represents the cause of the element preceding it.
+     * @param dispatchTask    the task that performs the actual dispatching.
+     * @return {@code true} if rescheduling succeeded, {@code false} if otherwise.
+     */
+    @Override
+    public boolean scheduleRetry(CommandMessage commandMessage,
+                                 RuntimeException lastFailure,
+                                 List<Class<? extends Throwable>[]> failures,
+                                 Runnable dispatchTask) {
+        int failureCount = failures.size();
+        if (!isExplicitlyNonTransient(lastFailure) && failureCount <= maxRetryCount) {
+            if (logger.isInfoEnabled()) {
+                logger.info("Processing of Command [{}] resulted in an exception. Will retry {} more time(s)... "
+                                    + "Exception was {}, {}",
+                            commandMessage.getPayloadType().getSimpleName(),
+                            maxRetryCount - failureCount,
+                            lastFailure.getClass().getName(),
+                            lastFailure.getMessage()
+                );
+            }
+            return scheduleRetry(dispatchTask, computeRetryInterval(commandMessage, lastFailure, failures));
+        } else {
+            if (failureCount >= maxRetryCount && logger.isInfoEnabled()) {
+                logger.info("Processing of Command [{}] resulted in an exception {} times. Giving up permanently. ",
+                            commandMessage.getPayloadType().getSimpleName(), failureCount, lastFailure);
+            } else if (logger.isInfoEnabled()) {
+                logger.info("Processing of Command [{}] resulted in an exception and will not be retried. ",
+                            commandMessage.getPayloadType().getSimpleName(), lastFailure);
+            }
+            return false;
+        }
+    }
+
+    /**
+     * A builder class for RetrySchedulers.
+     */
+    public static class Builder<B extends Builder> {
+
+        private ScheduledExecutorService retryExecutor;
+        private int maxRetryCount = DEFAULT_MAX_RETRIES;
+
+        protected void validate() throws AxonConfigurationException {
+            assertNonNull(retryExecutor, "The ScheduledExecutorService is a hard requirement and should be provided");
+            assertStrictPositive(maxRetryCount, "The maxRetryCount is a hard requirement and should be provided");
```

Comment:
The message in the `assertStrictPositive` seems incorrect to me. Shouldn't it state something along the lines of `The maxRetryCount is a strictly positive number`?

Reply:
Corrected